core/LSU.cpp

```diff
@@ -92,6 +95,11 @@ namespace olympia
              << ": "
              << memory_access_allocator_.getNumAllocated()
              << " MemoryAccessInfo objects allocated/created");
+
+        if(retire_done_and_is_drained_){
+            bool ldst_queue_empty_ = ldst_inst_queue_.empty();
+            sparta_assert(ldst_queue_empty_, "Issue queue has pending instructions");
```

Comment:
Probably can combine and drop the local variable:
```suggestion
            sparta_assert(ldst_inst_queue_.empty(), "Issue queue has pending instructions");
```